internal/gatewayapi/translator.go

```diff
@@ -777,6 +899,71 @@ func isValidCrossNamespaceRef(from crossNamespaceFrom, to crossNamespaceTo, refe
 	return false
 }
 
+// Checks if a hostname is valid according to RFC 1123 and gateway API's requirement that it not be an IP address
+func isValidHostname(hostname string) (valid bool, reason string) {
+	if len(hostname) == 0 || hostname == "" {
```

Comment:
Contour's using https://github.com/projectcontour/contour/blob/e067e1c59b44cd7b859b6ca8b958bd0018edc789/internal/gatewayapi/listeners.go#L161-L179 to do the valid hostname check (plus the same IP address check you have here) - if that upstream function works, we could use that in place of the custom validation code here.

Reply:
Re Luke's comments about `IsDNS1123Subdomain` since it does not check label length I think PR a fix for that in `IsDNS1123Subdomain` and then switch to using that when the change is accepted would be appropriate here :thinking:

Reply:
I've actually switched to using `IsDNS1123Subdomain` along with Arko's suggestion to include a check for individual label length.